gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -109,12 +132,23 @@ def self.signature_for_method(method)
   # Only public because it needs to get called below inside the replace_method blocks below.
   def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     current_declaration = T::Private::DeclState.current.active_declaration
-    return if !current_declaration
-    T::Private::DeclState.current.reset!
-
     mod = is_singleton_method ? hook_mod.singleton_class : hook_mod
     original_method = mod.instance_method(method_name)
 
+    if final_method?(original_method)
+      raise "`#{mod.name}##{method_name}` was declared as final and cannot be redefined"
```

Comment:
@DarkDimius Should we be calling `mod.name`? Do we have something in `sorbet-runtime` like what the constant cache is in `srb init`?